File: Source/JapaneseLanguageTools.Contracts/Models/Blobs/Responses/UploadBlobResponse.Model.cs
using System;

using JapaneseLanguageTools.Contracts.Models.Blobs.Requests;

namespace JapaneseLanguageTools.Contracts.Models.Blobs.Responses;

public class UploadBlobResponseModel
{
    public required BlobReferenceModel BlobReference { get; set; }

    public required DateTimeOffset BlobUploadedOn { get; set; }

    public required DateTimeOffset? BlobExpiresOn { get; set; }

    public required UploadBlobRequestModel Request { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/blobs/responses/UploadBlobResponse.Model.ts
import BlobReferenceModel from "@/models/blobs/BlobReference.Model";
import UploadBlobRequestModel from "@/models/blobs/requests/UploadBlobRequest.Model";

export default interface UploadBlobResponseModel {
  blobReference: BlobReferenceModel;
  blobUploadedOn: string;
  blobExpiresOn: string;

  request: UploadBlobRequestModel;
}
